cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs

```diff
@@ -635,7 +635,7 @@ runQueryPoolStateCmd
 
         let poolFilter = case allOrOnlyPoolIds of
               All -> Nothing
-              Only poolIds -> Just $ Set.fromList poolIds
+              Some poolId poolIds -> Just $ Set.fromList (poolId : poolIds)
```

Comment:
This has the same issue we spoke about previously. We can rely on `some` vs `many` here.